nearcache/config.go

```diff
@@ -0,0 +1,256 @@
+		return
+	}
+	c.evictionPolicy = &policy
+}
+
+// EvictionPolicy returns the eviction policy of this eviction configuration.
+// See the documentation for SetEvictionPolicy.
+func (c EvictionConfig) EvictionPolicy() EvictionPolicy {
+	if c.evictionPolicy == nil {
+		return defaultEvictionPolicy
+	}
+	return *c.evictionPolicy
+}
+
+// SetSize sets the number of maximum entries before an eviction occurs.
+// Accepts any non-negative number.
+// The default value is 10_000.
+func (c *EvictionConfig) SetSize(size int) {
+	s, err := check.NonNegativeInt32(size)
+	if err != nil {
+		c.err = ihzerrors.NewInvalidConfigurationError("nearcache.EvictionConfig.SetSize: out of range", err)
+		return
+	}
+	c.size = &s
+}
+
+// Size returns the number of maximum entries before an eviction occurs.
+// See the documentation for SetSize.
+func (c EvictionConfig) Size() int {
+	if c.size == nil {
+		return defaultMaxEntryCount
+	}
+	return int(*c.size)
+}
+
+// SetComparator sets the eviction policy comparator.
+func (c *EvictionConfig) SetComparator(cmp EvictionPolicyComparator) {
+	c.comparator = cmp
+}
+
+// Comparator returns the eviction policy comparator.
+func (c EvictionConfig) Comparator() EvictionPolicyComparator {
+	return c.comparator
+}
+
+// PreloaderConfig is the configuration for storing and pre-loading Near Cache keys.
+// Preloader re-populates Near Cache after client restart to provide fast access.
+// Saved preloader data is compatible between only the same versions of Go client.
+// It is disabled by default.
+type PreloaderConfig struct {
+	// Directory is the directory to store preloader cache.
+	Directory string
+	// StoreInitialDelaySeconds is the time in seconds before the preloader starts saving the cache.
+	// Must be positive.
+	// By default it is 600 seconds.
+	StoreInitialDelaySeconds int
+	// StoreIntervalSeconds is the time in seconds for the cache save period.
+	// Must be positive.
+	// By default it is 600 seconds.
+	StoreIntervalSeconds int
+	// Enabled enables the preloader.
+	Enabled bool
+}
+
+func (c PreloaderConfig) Clone() PreloaderConfig {
+	return c
+}
+
+func (c *PreloaderConfig) Validate() error {
+	if c.StoreInitialDelaySeconds == 0 {
+		c.StoreInitialDelaySeconds = defaultStoreInitialDelaySeconds
+	}
+	if c.StoreIntervalSeconds == 0 {
+		c.StoreIntervalSeconds = defaultStoreIntervalSeconds
+	}
+	if _, err := check.NonNegativeInt32(c.StoreInitialDelaySeconds); err != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.PreloaderConfig.StoreInitialDelaySeconds must be positive", nil)
+	}
+	if _, err := check.NonNegativeInt32(c.StoreIntervalSeconds); err != nil {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.PreloaderConfig.StoreIntervalSeconds must be positive", nil)
+	}
+	if c.Enabled && c.Directory == "" {
+		return ihzerrors.NewInvalidConfigurationError("nearcache.PreloaderConfig.Directory must not be empty if preloader is enabled.", nil)
+	}
```

Comment:
Does this logic exist in Java?

Reply:
Removed it at: https://github.com/hazelcast/hazelcast-go-client/pull/786/commits/112f3ac833c9d97eb9261bb3b436431ac8177081